src/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -141,6 +145,11 @@ internal AuthenticationResult(MsalAccessTokenCacheItem msalAccessTokenCacheItem,
         /// </summary>
         public IEnumerable<string> Scopes { get; }
 
+        /// <summary>
+        /// Gets the correlation id used for the request.
+        /// </summary>
+        public string CorrelationID { get; }
```

Comment:
nit:  CorrelationId  (.net naming conventions have ID as Id)

Do we want this to be TelemetryCorrelationId or is this more broad.  We're calling it TelemetryCorrelationId elsewhere.   
Should this also be a Guid type since that's how we're sending it in?

Reply:
Yes. it will be a Guid.
I think CorrelationId is sufficient on its own as it is a more general term.
@henrik-me @jmprieur thoughts?

Reply:
CorrelationId, is the right term imo. can be telemetry or not.